great_expectations/optional_imports.py

```diff
@@ -99,33 +99,211 @@ def is_version_less_than(
+    )
+    from sqlalchemy.sql.expression import (
+        BinaryExpression as sa_sql_expression_BinaryExpression,
+    )
+    from sqlalchemy.sql.expression import (
+        BooleanClauseList as sa_sql_expression_BooleanClauseList,
+    )
+    from sqlalchemy.sql.expression import (
+        ColumnClause as sa_sql_expression_ColumnClause,
+    )
+    from sqlalchemy.sql.expression import (
+        Label as sa_sql_expression_Label,
+    )
+    from sqlalchemy.sql.expression import (
+        Select as sa_sql_expression_Select,
+    )
+    from sqlalchemy.sql.expression import (
+        Selectable as sa_sql_expression_Selectable,
+    )
+    from sqlalchemy.sql.expression import (
+        TableClause as sa_sql_expression_TableClause,
+    )
+    from sqlalchemy.sql.expression import (
+        TextualSelect as sa_sql_expression_TextualSelect,
+    )
+    from sqlalchemy.sql.expression import (
+        WithinGroup as sa_sql_expression_WithinGroup,
+    )
+    from sqlalchemy.sql.operators import custom_op as sqlalchemy_custom_op
 except ImportError:
     sqlalchemy = SQLALCHEMY_NOT_IMPORTED
+    sqlalchemy_dialects_registry = SQLALCHEMY_NOT_IMPORTED
+    sqlalchemy_engine_Dialect = SQLALCHEMY_NOT_IMPORTED
+    sqlalchemy_engine_Inspector = SQLALCHEMY_NOT_IMPORTED
+    sqlalchemy_reflection = SQLALCHEMY_NOT_IMPORTED
+    sqlalchemy_engine_DefaultDialect = SQLALCHEMY_NOT_IMPORTED
+    sqlalchemy_DatabaseError = SQLALCHEMY_NOT_IMPORTED
+    sqlalchemy_IntegrityError = SQLALCHEMY_NOT_IMPORTED
+    sqlalchemy_NoSuchTableError = SQLALCHEMY_NOT_IMPORTED
+    sqlalchemy_OperationalError = SQLALCHEMY_NOT_IMPORTED
+    sqlalchemy_ProgrammingError = SQLALCHEMY_NOT_IMPORTED
+    SQLAlchemyError = SQLALCHEMY_NOT_IMPORTED
+    sa_sql_Insert = SQLALCHEMY_NOT_IMPORTED
+    sqlalchemy_literal = SQLALCHEMY_NOT_IMPORTED
+    quoted_name = SQLALCHEMY_NOT_IMPORTED
+    sa_sql_expression_CTE = SQLALCHEMY_NOT_IMPORTED
+    sa_sql_expression_BinaryExpression = SQLALCHEMY_NOT_IMPORTED
+    sa_sql_expression_BooleanClauseList = SQLALCHEMY_NOT_IMPORTED
+    sa_sql_expression_ColumnClause = SQLALCHEMY_NOT_IMPORTED
+    sa_sql_expression_Label = SQLALCHEMY_NOT_IMPORTED
+    sa_sql_expression_Select = SQLALCHEMY_NOT_IMPORTED
+    sa_sql_expression_Selectable = SQLALCHEMY_NOT_IMPORTED
+    sa_sql_expression_TableClause = SQLALCHEMY_NOT_IMPORTED
+    sa_sql_expression_TextualSelect = SQLALCHEMY_NOT_IMPORTED
+    sa_sql_expression_WithinGroup = SQLALCHEMY_NOT_IMPORTED
+    sqlalchemy_custom_op = SQLALCHEMY_NOT_IMPORTED
 
 try:
-    sqlalchemy_Connection = sqlalchemy.engine.Connection
-except (ImportError, AttributeError):
-    sqlalchemy_Connection = SQLALCHEMY_NOT_IMPORTED
+    from sqlalchemy.engine import (
+        Connection as sqlalchemy_engine_Connection,
+    )
+except ImportError:
+    sqlalchemy_engine_Connection = SQLALCHEMY_NOT_IMPORTED
 
 try:
-    sqlalchemy_Engine = sqlalchemy.engine.Engine
-except (ImportError, AttributeError):
```

Comment:
We need to keep these attribute errors

Reply:
@Kilo59 Could you please point to where in the new code it should be?  I am guessing line 172 and every other case of the "dot" access off `sqlalchemy` -- is this what you mean?  Thanks.